app/controllers/AuthenticationController.scala

```diff
@@ -406,6 +445,133 @@ class AuthenticationController @Inject()(
+          .user(userIdentity)
+          .timeout(60000)
+          .authenticatorSelection(
+            AuthenticatorSelectionCriteria.builder().residentKey(ResidentKeyRequirement.REQUIRED).build())
+          .build()
+        val registration = relyingParty.startRegistration(opts);
+        temporaryRegistrationStore.insert(request.identity._multiUser, registration);
+        Ok(Json.toJson(registration.toCredentialsCreateJson))
+      }(blockingContext)
+    } yield result;
+  }
+
+  def webauthnRegisterFinalize(): Action[WebAuthnRegistration] =
+    sil.SecuredAction.async(validateJson[WebAuthnRegistration]) { implicit request =>
+      {
+        val creationOpts = temporaryRegistrationStore.get(request.identity._multiUser)
+        temporaryRegistrationStore.remove(request.identity._multiUser)
+        creationOpts match {
+          case Some(data) => {
+            val response = PublicKeyCredential.parseRegistrationResponseJson(request.body.key);
+            val opts = FinishRegistrationOptions.builder().request(data).response(response).build();
+            try {
+              for {
+                preKey <- Fox.future2Fox(Future { Try(relyingParty.finishRegistration(opts)) })(blockingContext);
+                key <- preKey match {
+                  case scala.util.Success(key) => Fox.successful(key)
+                  case scala.util.Failure(e)   => Fox.failure("Registration failed", Full(e))
+                };
+                result <- {
+                  val credential = WebAuthnCredential(
+                    ObjectId.generate,
+                    request.identity._multiUser,
+                    WebAuthnCredentialRepository.byteArrayToBytes(key.getKeyId.getId),
+                    request.body.name,
+                    key.getPublicKeyCose.getBytes,
+                    key.getSignatureCount.toInt,
+                    isDeleted = false,
+                  )
+                  webAuthnCredentialDAO.insertOne(credential).map(_ => Ok(""))
+                }
+              } yield result;
+            } catch {
+              case e: RegistrationFailedException => Future.successful(BadRequest("Failed to register key"))
+            }
+          }
+          case None => Future.successful(BadRequest("Challenge not found or expired"))
+        }
+      }
+    }
+
+  def webauthnListKeys: Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    {
+      for {
+        keys <- webAuthnCredentialDAO.listKeys(request.identity._multiUser)
+        reducedKeys = keys.map(credential => WebAuthnKeyDescriptor(credential._id, credential.name))
+      } yield Ok(Json.toJson(reducedKeys))
+    }
+  }
+
+  def webauthnRemoveKey: Action[WebAuthnKeyDescriptor] = sil.SecuredAction.async(validateJson[WebAuthnKeyDescriptor]) {
+    implicit request =>
+      {
+        for {
+          _ <- webAuthnCredentialDAO.removeById(request.body.id, request.identity._multiUser)
+        } yield Ok(Json.obj())
+      }
+  }
```

Comment:
It is totally fine to pass the entry that should be deleted in the request body. But I think usually when we send only an ID, this is done via url params. But as your approach is totally fine, feel free to choose one. Just wanted to leave that thought here :D